### DISCUSSION TITLE
ORAN 7.2 Benetel RAN650 RU

### DISCUSSION THREAD

**Discussion by vamaglog** [srsran/srsRAN_Project#91] [General Help] [answered]
Hi,

First of all, it is really great news that 7.2 split is now supported by srsRAN_Project! :) Congratulations to the whole team for the great achievement. We are looking forward to start experimenting with it.

In our lab, we have a couple of Benetel RAN650 RU that are the outdoor version of the RAN550 used in the tutorial.

I saw that RAN650 is also mentioned in your news letter last February (https://srs.io/srs-introduces-new-oran-native-5g-cu-du-solution-at-mwc/). Hence, I am wondering if you have any experience with those RUs as well. 

Do you have any guidelines on how to connect the RAN650 to the DU? Is the procedure similar to the RAN550? Meaning, should the compression parameters be set as instructed in the tutorial and are the same registers for both RU types (c0350, c0351, c0352 and c0353)? 
Moreover, could you please explain how exactly we configure the registers c0350, c0351, c0352 to 0 and the register c0353 to 1? Probably by using the registercontrol and eeprog_cp60 commands, but could you please list them explicitly?

Finally, is there another modification in the example configuration file needed related to RAN650 RU?

Thank you very much in advance for the information.
Best regards,
Vasilis

**Comment by ismagom** [comment]
Hi Vasilis, 

Thanks for your feedback. 

I think RAN650 and RAN550 should be compatible from the point of view of interaction with DU and basic configuration. My best recommendation is to follow the "Software User Guide for RANxx0 CAT-A O-RUs" from Benetel, or contact them directly. 

Regarding those registers in particular, the manual doesn't seem to show any difference between RU models, so I think you could try following the same procedure. 

Please let us know how the integration works. 

Regards

**Comment by merimdzaferagic** [comment]
> Hi,
> 
> First of all, it is really great news that 7.2 split is now supported by srsRAN_Project! :) Congratulations to the whole team for the great achievement. We are looking forward to start experimenting with it.
> 
> In our lab, we have a couple of Benetel RAN650 RU that are the outdoor version of the RAN550 used in the tutorial.
> 
> I saw that RAN650 is also mentioned in your news letter last February (https://srs.io/srs-introduces-new-oran-native-5g-cu-du-solution-at-mwc/). Hence, I am wondering if you have any experience with those RUs as well.
> 
> Do you have any guidelines on how to connect the RAN650 to the DU? Is the procedure similar to the RAN550? Meaning, should the compression parameters be set as instructed in the tutorial and are the same registers for both RU types (c0350, c0351, c0352 and c0353)? Moreover, could you please explain how exactly we configure the registers c0350, c0351, c0352 to 0 and the register c0353 to 1? Probably by using the registercontrol and eeprog_cp60 commands, but could you please list them explicitly?
> 
> Finally, is there another modification in the example configuration file needed related to RAN650 RU?
> 
> Thank you very much in advance for the information. Best regards, Vasilis

Hi Vasilis, 

I am working on the same setup at the moment (with the RAN650). I believe that this command should help you set the registers: 
`registercontrol -w c0350 -x 0x01 >> /home/root radio_boot_response`

Regards, 
Merim

**Comment by mateuszmichalskiacc** [comment]
@vamaglog @merimdzaferagic 
Hi, 
I'm trying to do the same with RAN650. Does it work for you?
I've managed to have working DL broadcast and the UE can see the cell with SIB1. However I'm struggling with PRACH detection. In srsRAN logs I see that RACH is being processed in the DL slots instead of correct uplink slots. Strange.

@ismagom I have a question about TDD pattern which is mentioned in the srsRAN manual. There is 6-3 pattern motioned.
Does it mean the DDDDDD(S)UUU?
If yes, the special slots is  guard symbols only?

**Reply by ismagom** [reply]
Yes, the pattern DDDDDD(S)UUU and no special slots is configured with: 

```
  tdd_ul_dl_cfg:
    nof_dl_slots: 6
    nof_ul_slots: 3
``` 

if you want to use DL or UL special slots, just use eg: 

```
  tdd_ul_dl_cfg:
    nof_dl_slots: 7
    nof_dl_symbols: 6
    nof_ul_slots: 2
    nof_ul_symbols: 2
```

**Reply by merimdzaferagic** [reply]
@mateuszmichalskiacc 
Hi, 

Is the RU that you have Cat A or Cat B?

**Reply by mateuszmichalskiacc** [reply]
@merimdzaferagic It's CatA
@ismagom I still can't detect the PRACH by DU, however I see that there is power on OFH IQ data in the expected subframe and PRB.
Any idea how can I debug why DU can't detect PRACH? 
I've already compered the UE and OFH traces and the preamble is there... but still no detection by DU...

**Reply by ismagom** [reply]
Hi @mateuszmichalskiacc , can you send gnb.yml config and gnb.log? set `all_level: info`

thanks

**Reply by mateuszmichalskiacc** [reply]
@ismagom 
Sure, here they are.
[gnb.zip](https://github.com/srsran/srsRAN_Project/files/13296540/gnb.zip)
[gnb.yml.txt](https://github.com/srsran/srsRAN_Project/files/13296543/gnb.yml.txt)

**Reply by ismagom** [reply]
Hi @mateuszmichalskiacc, the first thing I spot is some potential issue with the PRACH configuration. At least with the RAN550 the only supported long formats configuration index were 17 and 19 and short formats 156 and 151. 

Have you checked with the Benetel user manual?

**Reply by mateuszmichalskiacc** [reply]
Hi @ismagom 
Yes I did check that. Which Benetel SW release are you talking about?

I'm using v0.8.1 and there is nothing about Long PRACH config index configuration in user manual.

However, I've tried couple of combinations and RU didn't behave as expected. It looks that on default Benetel's config the RU behaves like prach config index 27, which is my understanding some hack from RU side to support more prach configs. So prach config index = 7 should work.

**Reply by faluco** [reply, marked as answer]
Hi @mateuszmichalskiacc 

In the logs you have shared there are traces like this one:
```
Handling PRACH in slot 54.19: port=0, symbol=0
```
This log line gets printed each time the OFH library pushes the PRACH samples to the PRACH detector. I would suggest to enable phy layer logs to check what the detector sees, you can do this by enabling the `broadcast_enabled` in the log section of the config file and setting `phy_level` to debug.

We are using v0.8.1 for the RAN550, to enable the long prach format you have to run `long_form_prach.sh` inside the sbin dir in the RU or tweak the init radio script file. In this file there is a section to enable short prach, for our lab RU we have disabled this part to leave long prach format by default.

**Reply by faluco** [reply]
From your logs we can see:
```
2023-11-08T11:59:39.970290 [UL-PHY0 ] [I] [  564.18] PRACH:  rssi=-81.8dB detected_preambles=[{idx=2 ta=0.00us power=-8.1dB snr=-8.1dB}] t=196.5us
...
2023-11-08T11:59:39.971024 [SCHED   ] [I] [   565.7] Slot decisions cell=0 t=14us (1 PDSCH, 0 PUSCHs): RAR: ra-rnti=0x7f rb=[0..8) tbs=8
...
2023-11-08T11:59:42.890290 [UL-PHY0 ] [I] [  856.18] PRACH:  rssi=-82.0dB detected_preambles=[{idx=1 ta=0.00us power=-8.2dB snr=-8.2dB}] t=196.0us
...
2023-11-08T11:59:42.891015 [SCHED   ] [I] [   857.7] Slot decisions cell=0 t=9us (1 PDSCH, 0 PUSCHs): RAR: ra-rnti=0x7f rb=[0..8) tbs=8
```
The prach detector detected 2 prach preambles and then the corresponding RAR was sent to the UE, however notice that the rssi and snr is very low, so they are false alarms and the TA is probably calculated wrong.
Please make sure you have long prach format configured in the RU. Also you could try using only a single UL port for testing by setting `nof_antennas_ul` to 1 and `ul_port_id` to [0] and `prach_port_id` to [4].

**Comment by mateuszmichalskiacc** [comment]
Hi @faluco @ismagom 
It magically started working. We got some explanations from Benetel as well.
Basically after fresh Benetel SW install 0.8.1 the instructions provided by @faluco will work. 
Thanks for your help
